### PR TITLE
Add matrix representations to admin and schema

### PR DESCRIPTION
**Pull request by frwickst**
This is an initial version of matrix representations.
The admin integration is rudimentary at best and does not include any proper validation. Currently one can create a matrix when a section has at least 4 section attributes. Note that there should _not_ be more than one unique section attribute per matrix, even if the admin interface might allow it for now.

### CODE REVIEW

File: projects/admin.py [hunk 1]
```diff
@@ -125,3 +129,71 @@ class DocumentTemplateAdmin(admin.ModelAdmin):
...
+        self, request, context, add=False, change=False, form_url="", obj=None
+    ):
+        cell_rows = []
+        if obj:
+            # The matrix is square so we just need one side to know the other
+            side_length = len(obj.column_names)
+
+            # Fill matrix with None values
+            cell_rows = [
+                [None for x in range(0, side_length)] for y in range(0, side_length)
```
Comment (charn) [inline projects/admin.py]: `[None] * side_length` should be equal to `[None for x in range(0, side_length)]`

Comment (charn) [reply projects/admin.py]: Also `list(range(0,5)) == list(range(5))`

Comment (charn) [reply projects/admin.py]: So the short version of this would be: `[[None] * side_length] * side_length`

File: projects/admin.py [hunk 2]
```diff
@@ -125,3 +129,71 @@ class DocumentTemplateAdmin(admin.ModelAdmin):
...
+@admin.register(PhaseAttributeMatrixStructure)
+class PhaseAttributeMatrixStructureAdmin(admin.ModelAdmin):
+    change_form_template = "admin/matrix.html"
+
+    def render_change_form(
+        self, request, context, add=False, change=False, form_url="", obj=None
+    ):
+        cell_rows = []
+        if obj:
+            # The matrix is square so we just need one side to know the other
```
Comment (charn) [inline projects/admin.py]: Can't we have different number of rows ans columns?

File: projects/templatetags/list_tools.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from django import template
+
+register = template.Library()
+
+
+@register.filter
+def index(List, i):
```
Comment (charn) [inline projects/templatetags/list_tools.py]: I guess the name `List` is used here to not overwrite the built-in `list`? `list_object` would also work. :thinking:

Comment (frwickst) [reply projects/templatetags/list_tools.py]: Yeah, this is a mistake from my side, should not be `List` for sure :)

File: projects/admin.py [hunk 4]
```diff
@@ -125,3 +129,69 @@ class DocumentTemplateAdmin(admin.ModelAdmin):
...
+class PhaseAttributeMatrixStructureAdmin(admin.ModelAdmin):
+    change_form_template = "admin/matrix.html"
+
+    def render_change_form(
+        self, request, context, add=False, change=False, form_url="", obj=None
+    ):
+        cell_rows = []
+        if obj:
+            x_range = range(len(obj.row_names))
+            y_range = range(len(obj.column_names))
```
Comment (charn) [inline projects/admin.py]: I think column names is the horizontal, which I think is usually the `x`

File: projects/templatetags/list_tools.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from django import template
+
+register = template.Library()
+
+
+@register.filter
+def index(l: list, i: str):
```
Comment (charn) [inline projects/templatetags/list_tools.py]: :+1:

File: projects/admin.py (hunk 4 above)
Comment (frwickst) [reply projects/admin.py]: wups :D

Comment (frwickst) [reply projects/admin.py]: Was wondering why I had to reverse the order of the`cell_row` variable I was creating as well 😄

**Pull request merged**